### PR TITLE
BUG: error raise when column contains percentage

### LINKED ISSUE #37157: BUG: pd.read_sql_table() raises unknown column error when column name of a table contains `%`
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

###### Mysql Create Table
```
DROP TABLE IF EXISTS test.my_table;
CREATE TABLE test.my_table 
(
    `id` VARCHAR(3),
    PRIMARY KEY(id),
    `price` DOUBLE NULL,
    `%_variation` DOUBLE NULL
);

INSERT INTO test.my_table VALUES('101', 25.2, 0.1);
INSERT INTO test.my_table VALUES('102', 40.2, -0.5);
INSERT INTO test.my_table VALUES('103', 55.2, 0.9);
```

```python
import pandas as pd
from sqlalchemy import create_engine

DB_PW = 'My_password'
DB_PORT = '3306'
DB_NAME = 'Test'

db_uri = f'mysql+mysqldb://root:{DB_PW}@localhost:{DB_PORT}'
engine = create_engine(f'{db_uri}/{DB_NAME}')

df = pd.read_sql_table(table_name='my_table', con=engine, schema=DB_NAME)
```

#### Problem description

[this should explain **why** the current behaviour is a problem and why the expected output is a better solution]
image

image

As you can see above, the `%_variation` column is read as `%%_variation` in the SELECT statement. <br> 
But the `%%_variation` column is not present in the database and thus reading this table from the database provides an error.

#### Expected Output

I expect pandas to read the table from the database. <br>

I tried reading with pandas version 1.0.5 and I was able to read the table from the database without any problem. <br>
image

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.7.7.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.3
numpy            : 1.19.0
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.3
setuptools       : 50.3.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : 1.3.19
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by erfannariman**
- [x] closes #37157 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by erfannariman** [comment]
I think this solves #11896 as well? Not 100% sure

**Review by arw2019 — COMMENTED** [review]
lgtm but needs to be looked over by someone who's more familiar with the sql module

@erfannariman would you mind merging master again to make sure this up to date

**Comment by erfannariman** [comment]
> lgtm but needs to be looked over by someone who's more familiar with the sql module
> 
> @erfannariman would you mind merging master again to make sure this up to date

Yes will do asap.

**Comment by jreback** [comment]
this was deliberately changed in this was deliberately changed in #34212 so not sure why you think this is a regression.

File: doc/source/whatsnew/v1.1.5.rst [hunk 1]
```diff
@@ -29,6 +29,7 @@ Fixed regressions
 - Fixed performance regression in ``df.groupby(..).rolling(..)`` (:issue:`38038`)
 - Fixed regression in :meth:`MultiIndex.intersection` returning duplicates when at least one of the indexes had duplicates (:issue:`36915`)
 - Fixed regression in :meth:`.GroupBy.first` and :meth:`.GroupBy.last` where ``None`` was considered a non-NA value (:issue:`38286`)
+- Fixed regression in :func:`read_sql_table` raising a ``sqlalchemy.exc.OperationalError`` when column names contained a percentage sign (:issue:`37517`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.5.rst:32]: move to 1.2 i think we could put this in.

**Review by jreback — CHANGES_REQUESTED** [review]
ok i c, this doesn't rever the prior just fixes an issue.

File: doc/source/whatsnew/v1.1.5.rst (hunk 1 above)
Comment (erfannariman) [reply doc/source/whatsnew/v1.1.5.rst]: Moved it to 1.2 and as bug fix, since it's not a regression fix if I understood you correctly.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @erfannariman

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**